src/System.Linq.Expressions/tests/ExceptionHandling/ExceptionHandlingExpressions.cs

```diff
@@ -137,6 +137,108 @@ public void CanCatchAndThrowNonExceptionsInterpreted()
         }
 
         [Fact]
+        public void ThrowNullThrowsNRE()
+        {
+            Action throwNull = Expression.Lambda<Action>(
+                Expression.Throw(Expression.Constant(null, typeof(Expression)))
+                ).Compile(false);
+            Assert.Throws<NullReferenceException>(throwNull);
+        }
+
+        [Fact]
+        public void ThrowNullThrowsNREInterpreted()
+        {
+            Action throwNull = Expression.Lambda<Action>(
+                Expression.Throw(Expression.Constant(null, typeof(Expression)))
+                ).Compile(true);
+            Assert.Throws<NullReferenceException>(throwNull);
+        }
```

Comment:
That might be better. There are tests, including in this file, where the one works and one relates to an active issue, which makes it a bit less consistent, but I could always make exceptions just for those.